File: CS/Solution28.Web/CodeCentralExampleInMemoryDataStoreProvider.cs
using System;
using System.Data;
using DevExpress.Xpo.DB;

namespace E2813.Demo.Web {
    #region For demo purposes only!!!
    public class CodeCentralExampleInMemoryDataStoreProvider {
        private static readonly string fConnectionString;
        private static readonly DataSet fdataSet;
        public static string ConnectionString { get { return fConnectionString; } }
        static CodeCentralExampleInMemoryDataStoreProvider() {
            string providerKey = Guid.NewGuid().ToString();
            fConnectionString = String.Format("XpoProvider={0}", providerKey);
            fdataSet = new DataSet();
            DataStoreBase.RegisterDataStoreProvider(providerKey, CreateProviderFromString);
        }
        public static IDataStore CreateProviderFromString(string connectionString, AutoCreateOption autoCreateOption, out IDisposable[] objectsToDisposeOnDisconnect) {
            objectsToDisposeOnDisconnect = new IDisposable[] { };
            return new DataSetDataStore(fdataSet, autoCreateOption);
        }
    }
    #endregion
}

File: VB/Solution28.Web/CodeCentralExampleInMemoryDataStoreProvider.vb
Imports System
Imports System.Data
Imports DevExpress.Xpo.DB
Imports System.Runtime.InteropServices

Namespace E2813.Demo.Web

'#Region "For demo purposes only!!!"
    Public Class CodeCentralExampleInMemoryDataStoreProvider

        Private Shared ReadOnly fConnectionString As String

        Private Shared ReadOnly fdataSet As DataSet

        Public Shared ReadOnly Property ConnectionString As String
            Get
                Return fConnectionString
            End Get
        End Property

        Shared Sub New()
            Dim providerKey As String = Guid.NewGuid().ToString()
            fConnectionString = String.Format("XpoProvider={0}", providerKey)
            fdataSet = New DataSet()
            Call DataStoreBase.RegisterDataStoreProvider(providerKey, New Helpers.DataStoreCreationFromStringDelegate(AddressOf CreateProviderFromString))
        End Sub

        Public Shared Function CreateProviderFromString(ByVal connectionString As String, ByVal autoCreateOption As AutoCreateOption, <Out> ByRef objectsToDisposeOnDisconnect As IDisposable()) As IDataStore
            objectsToDisposeOnDisconnect = New IDisposable() {}
            Return New DataSetDataStore(fdataSet, autoCreateOption)
        End Function
    End Class
'#End Region
End Namespace
